src/modules/tests/test-data/cat-adi-part2/vehicle-checks/vehicle-checks.cat-adi-part2.reducer.ts

```diff
@@ -4,26 +4,60 @@ import {
   NUMBER_OF_TELL_ME_QUESTIONS as numberOfTellMeQuestions,
 }
 from '../../../../../shared/constants/tell-me-questions/tell-me-questions.cat-adi-part2.constants';
-import {
-  NUMBER_OF_SHOW_ME_QUESTIONS as numberOfShowMeQuestions,
-}
-from '../../../../../shared/constants/show-me-questions/show-me-questions.cat-adi-part2.constants';
+import { CompetencyOutcome } from '../../../../../shared/models/competency-outcome';
 
 export const initialState: CatADI2UniqueTypes.VehicleChecks = {
   tellMeQuestions: Array(numberOfTellMeQuestions).fill({}),
-  showMeQuestions: Array(numberOfShowMeQuestions).fill({}),
+  showMeQuestions: [
+    {
+      code: '',
+      description: '',
+      outcome: CompetencyOutcome.P,
+    },
+    {
+      code: '',
+      description: '',
+      outcome: CompetencyOutcome.P,
+    },
+  ],
 };
 
 export function vehicleChecksCatADI2Reducer(
   state: CatADI2UniqueTypes.VehicleChecks = initialState,
   action: vehicleChecksCatADI2ActionTypes.Types,
 ): CatADI2UniqueTypes.VehicleChecks {
+  let showMeQuestionIndex;
+  let showMeFaultCount = 0;
+  if (state.showMeQuestions) {
```

Comment:
as said previously, im not sure if this logic should exist in the reducer

Reply:
i could move all the logic inside SHOW_ME_QUESTION_DRIVING_FAULT, not sure where else we'd preform this.  That part of the reducer needs to know how many showme and tellme driving faults there are in order to fire. And this code is getting that from the state and updating the state appropriately.  It may not be the cleanest implementation, but its using the state to update the state if needed.

Reply:
re-written